torch/csrc/utils/auto_cuda_event.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+// RAII struct to to record and wait on CUDA events
+
+#ifdef USE_CUDA
+
+#include "cuda.h"
+#include "cuda_runtime.h"
+#include <ATen/ATen.h>
+#include <THC/THC.h>
+
+struct AutoCUDAEvent {
+  AutoCUDAEvent() = default;
+  ~AutoCUDAEvent() {
+    if (is_created) THCudaCheck(cudaEventDestroy(event));
+  }
+
+  void record(THCStream* stream) {
+    if (!is_created) {
+      THCudaCheck(cudaEventCreateWithFlags(&event, cudaEventDisableTiming));
+      THCudaCheck(cudaEventRecord(event, THCStream_stream(stream)));
+      is_created = true;
+    } 
+  }
+
+  void wait_on(THCStream* stream) {
```

Comment:
This is a bit misleading beacuse `evt.wait_on(stream)` makes `stream` wait on `evt`. How about `evt.block(stream)`?

Reply:
Sounds good.